feat(replay): viewed_by_me search alias

aliu39 opened the pull request:
Relates to https://github.com/getsentry/sentry/issues/64924

Frontend change coming soon
I'm ok with removing the `viewed_by_id` parameter from the typeahead. Thoughts?

aliu39 committed:
feat(replay): viewed_by_me search alias

aliu39 committed:
feat(replay): viewed_by_me search alias

cmanallen commented on src/sentry/replays/usecases/query/__init__.py:
```diff
@@ -163,6 +165,17 @@ def query_using_optimized_search(
             SearchFilter(SearchKey("environment"), "IN", SearchValue(environments)),
         ]
 
+    for i in range(len(search_filters)):
```
`for search_filter in search_filters:`

cmanallen commented on src/sentry/replays/usecases/query/__init__.py:
```diff
@@ -163,6 +165,17 @@ def query_using_optimized_search(
+                raise ParseError("Invalid operator specified for viewed_by_me")
+            search_filters[i] = SearchFilter(
+                SearchKey("viewed_by_id"), sf.operator, SearchValue(request_user_id)
+            )
```
How is this handled `viewed_by_me:false`?

aliu3ntry replied:
Oh, good point!

cmanallen commented on src/sentry/replays/usecases/query/__init__.py:
```diff
@@ -163,6 +165,17 @@ def query_using_optimized_search(
+            search_filters[i] = SearchFilter(
+                SearchKey("viewed_by_id"), sf.operator, SearchValue(request_user_id)
+            )
+
```
Needs coverage.  Both fields, true and false values, `=` and `!=` operators.

cmanallen commented on src/sentry/replays/usecases/query/__init__.py:
```diff
@@ -163,6 +165,17 @@ def query_using_optimized_search(
+            # "viewed_by_me" is not a valid query field.
+            # It's a convenience alias for users without the admin access to lookup ids.
+            if sf.operator not in ["=", "!="]:
+                raise ParseError("Invalid operator specified for viewed_by_me")
```
How do you know the user queried by `viewed_by_me`?  Could have queried by `seen_by_me`.

cmanallen commented on src/sentry/replays/usecases/query/__init__.py:
```diff
@@ -163,6 +165,17 @@ def query_using_optimized_search(
+            if sf.operator not in ["=", "!="]:
+                raise ParseError("Invalid operator specified for viewed_by_me")
+            search_filters[i] = SearchFilter(
+                SearchKey("viewed_by_id"), sf.operator, SearchValue(request_user_id)
```
Even though it sholuldn't be possible we need to handle the case where the user_id is null.  Unless you want to enforce non-nullability of the user in the call site (which would also be a fine outcome).

evanh committed:
fix: Add a metric for non-success Snuba requests (#70452)

Tracking this on the Sentry side allows alerts to be created that are
separate
from the Snuba API itself, in case the API is in a broken state and
can't
accurately report what is happening.

aliu39 committed:
Handle value=false

aliu39 committed:
Handle value=false

aliu39 committed:
Handle alias for exception msg

aliu39 committed:
Handle alias for exception msg

aliu3ntry commented on src/sentry/replays/usecases/query/__init__.py:
```diff
@@ -163,6 +165,18 @@ def query_using_optimized_search(
+            # It's a convenience alias for users without the admin access to lookup ids.
+            if sf.operator not in ["=", "!="]:
+                raise ParseError("Invalid operator specified for viewed_by_me")
+            is_me = not ((sf.operator == "=") ^ sf.value.value)
```
Got kinda fancy with it. Lmk if this is unreadable lol

aliu39 committed:
Comment explaining xor

aliu39 committed:
Comment explaining xor

evanpurkhiser committed:
ref(routes): A few more routes using withOrgPath (#70449)

colin-sentry committed:
chore(ai-monitoring): Add a unit to total cost (#70484)

mrduncan committed:
chore(issues): Enable stronger typing on occurrence_consumer (#70487)

This is a quick follow up to #69828 since these two type errors are
trivial to resolve.

cmanallen commented on src/sentry/replays/usecases/query/__init__.py:
```diff
@@ -163,6 +165,20 @@ def query_using_optimized_search(
+                raise ParseError(f"Invalid operator specified for {sf.key.name}")
+            is_me = not (
+                (sf.operator == "=") ^ sf.value.value
+            )  # True if both expressions are the same, else False.
```
Very fancy!  Let's change this though.  Only the `=` operator should be allowed.  I think `!=` doesn't make sense in this context.  Now we only need to care about the value and can drop the xor.

aliu3ntry replied:
The problem is the frontend search bar: users don't type in `key=val` or `key!=val`, they type `key:val` and `!key:val`. The latter translates into the != operator. So unless we want to drop searches for `!viewed_by_me:x`, we have to handle !=...

aliu3ntry replied:
I guess if users know its a boolean, it'd be pretty weird to do `!viewed_by_me:true`. So we can just raise ParseError to simplify this..

cmanallen commented on src/sentry/replays/usecases/query/__init__.py:
```diff
@@ -163,6 +165,20 @@ def query_using_optimized_search(
+                (sf.operator == "=") ^ sf.value.value
+            )  # True if both expressions are the same, else False.
+            search_filters[i] = SearchFilter(
+                SearchKey("viewed_by_id"), "=" if is_me else "!=", SearchValue(request_user_id)
```
Let's assign this inline `if` to a variable `operator`.  A little more verbose but it will lower the density of this code.

cmanallen commented on src/sentry/replays/usecases/query/__init__.py:
```diff
@@ -163,6 +165,20 @@ def query_using_optimized_search(
+            # "viewed_by_me" is not a valid query field.
+            # It's a convenience alias for users without the admin access to lookup ids.
+            if sf.operator not in ["=", "!="]:
+                raise ParseError(f"Invalid operator specified for {sf.key.name}")
```
Also need to assert the value is a bool.

ceorourke committed:
ref(daily summary): Disable notification (#70295)

I don't currently have any time to dedicate to bug fixing and it's been
sending multiple times, so this is going to be disabled for now. We'll
likely have to keep track of which users per org received the
notification and check that before sending, and then clear it out every
hour to avoid duplicate sending.

evanpurkhiser committed:
ref(crons): Use constant for DEFAULT_CHECKIN_MARGIN (#70510)

evanpurkhiser committed:
ref(crons): Move timezone to schedule text (#70511)

<img width="295" alt="image"
src="https://github.com/getsentry/sentry/assets/1421724/fd919e5b-0dd6-4b41-8e84-d3b99dfe6042">

It is now next to the schedule. This is more logical

mrduncan committed:
chore(issues): Enable stronger typing on two endpoints (#70488)

This is another quick follow up to #69828 since these two type errors
are trivial to resolve.

evanpurkhiser committed:
ref(crons): Improve details legend (#70515)

The legend now differentiates between timeout and failed

<img alt="clipboard.png" width="267"
src="https://i.imgur.com/yQAfXaB.png" />

---------

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

aliu39 committed:
Disallow !=

aliu39 committed:
Disallow !=

smeubank committed:
Add Deno Runtime Icon (#69100)

QoL, this has been bugging me


Should show the Deno icon, from Deno runtimes hosted on Supabase and
others

after:

![image](https://github.com/getsentry/sentry/assets/47563310/a1c63d17-3a89-4639-93d1-bca250f0bc31)


![image](https://github.com/getsentry/sentry/assets/47563310/ca66d7cd-8315-4366-a828-8a93936c1a07)

c298lee committed:
ref(replay): Rage click clicked element name (#70493)

Since we now use react component names in the selector path, we should
modify clicked element to provide more specific info such as class, role
etc.
This prevents giving the react component name twice and would give more
info for debugging since react component names aren't very specific

Before: 
<img width="845" alt="image"
src="https://github.com/getsentry/sentry/assets/55311782/87b6c8cd-1b29-40bb-b448-715595cb255b">

iamrajjoshi committed:
feat(chartcuterie): Change Chart Stylings (#70489)

two quick updates:

1. Changes the label for regressed to a darker color so we can read it
better

![image](https://github.com/getsentry/sentry/assets/33237075/80cd09dd-3619-4d58-abd3-65795de248c0)

2. Updated some chart styling for slack so the legend is on the left 

![image](https://github.com/getsentry/sentry/assets/33237075/de5a048a-cad5-4e9a-b152-15cbf9bc254e)

k-fish committed:
feat(traces): More improvements, use sdk names now (#70514)

### Summary
This uses sdk names as well as project to differentiate when it comes to
multi-service single project traces. This also adds some small UI
tweaks.

michellewzhang committed:
fix(replay): check for undefined property in init crumb (#70500)

fixes JAVASCRIPT-2SZ9
fixes JAVASCRIPT-2T0N

check for undefined properties before we attempt to access them. the
replay init breadcrumb doesn't have `data.to`:


https://github.com/getsentry/sentry/blob/257a223719efcfc4d77e4440e3f3e30e88fbbf88/static/app/utils/replays/hydrateBreadcrumbs.tsx#L59-L70

asottile-sentry committed:
ref: fix typing of slack test_requests (#70507)

<!-- Describe your PR here. -->

aliu39 committed:
(WIP) add tests

aliu39 committed:
Merge branch 'aliu/viewed-by-me' of github.com:getsentry/sentry into aliu/viewed-by-me

aliu3ntry commented:
Sorry all... my rebase git config keeps resetting 🤦‍♂️

MichaelSun48 commented:
Happens to the best of us... 

`git push origin <branch> --force` after a local rebase worked as expected for me before

aliu3ntry commented:
> Happens to the best of us...
> 
> `git push origin <branch> --force` after a local rebase worked as expected for me before

Thanks..
Nah, I've gone over to the merge side. No rebasing 🙅‍♂️
